src/main/java/org/digma/intellij/plugin/service/InsightsActionsService.java

```diff
@@ -44,4 +48,14 @@ public void navigateToMethod(@NotNull String codeObjectId) {
     public void openWorkspaceFileForSpan(@NotNull String workspaceUri, int offset) {
         editorService.openSpanWorkspaceFileInEditor(workspaceUri, offset);
     }
+
+    public void refreshInsightsList() {
+        Scope scope = insightsViewService.getModel().getCurrentScope();
+        if(scope instanceof MethodScope) {
+            MethodInfo methodInfo = ((MethodScope) scope).getMethodInfo();
+            Backgroundable.ensureBackground(project, "Refresh insights list",
+                    () -> insightsViewService.contextChanged(methodInfo)
```

Comment:
I may be wrong, but it seems that `InsightsViewService` already has the current `MethodInfo`:
`((MethodScope)InsightsViewService.model.scope).getMethodInfo()`

So maybe the new `PanelModel.getCurrentScope()` is not needed? And we can achieve the same results by adding (and calling) a new method like this:
``` kotlin
class InsightsViewService(project: Project) : AbstractViewService(project) {
...
    fun refreshInsights(){
        val scope = model.scope
        if (scope is MethodScope)
        {
            contextChanged(scope.getMethodInfo())
        }
    }
...
}
```

Reply:
Nice catch! Let me try with this approach :)

Reply:
Implemented by your approach! thanks!